dev/io.openliberty.microprofile.config.2.0.internal_fat/fat/src/io/openliberty/microprofile/config/internal_fat/apps/defaultSources/DefaultSourcesTestServlet.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.config.internal_fat.apps.defaultSources;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import javax.servlet.annotation.WebServlet;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.junit.Assert;
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+
+@WebServlet("/")
+public class DefaultSourcesTestServlet extends FATServlet {
+
+    /**
+     *
+     * @throws Exception
+     */
+    @Test
+    public void defaultSystemConfigProperties() throws Exception {
+        Map<String, String> systemValues = new HashMap<>(System.getenv());
+        Properties props = System.getProperties();
+
+        // Properties override environment variables.
+        for (Map.Entry<?, ?> entry : props.entrySet()) {
+            String key = (String) entry.getKey();
+            String value = (String) entry.getValue();
+            systemValues.put(key, value);
+        }
+
+        ConfigBuilder builder = ConfigProviderResolver.instance().getBuilder();
+        builder.addDefaultSources();
+        Config config = builder.build();
+
+        // Bit of a naughty cast but we know it's a set and that's easier to work on
+        Set<String> configPropertyNames = (Set<String>) config.getPropertyNames();
+
+        Assert.assertFalse("There should be at least 1 system value.", systemValues.isEmpty());
+
+        // Check expected Config property values are present
+        for (Map.Entry<String, String> systemValue : systemValues.entrySet()) {
+
+            String expectedKey = systemValue.getKey();
+
+            Assert.assertTrue("The system value for " + expectedKey + " should be a Config property name",
+                              configPropertyNames.contains(expectedKey));
+
+            if (!systemValue.getValue().isEmpty()) {
+
+                String expectedValue = systemValue.getValue();
+                String actualValue = config.getValue(expectedKey, expectedKey.getClass());
+
+                Assert.assertTrue("The system value for " + expectedKey + " should be " + expectedValue + ", not " + actualValue,
+                                  expectedValue.equals(actualValue));
+            }
+        }
```

Comment:
This test isn't correct any more if any of the system properties or environment variables contain `${foo}` as this would trigger property expansion so the input value won't match the value retrieved from config.

This does seem to happen on some linux systems.